analyzer/codechecker_analyzer/cmd/checkers.py

```diff
@@ -202,217 +206,273 @@ def add_arguments_to_parser(parser):
     parser.set_defaults(func=main)
 
 
-def main(args):
+def __uglify(text):
     """
-    List the checkers available in the specified (or all supported) analyzers
-    alongside with their description or enabled status in various formats.
+    csv and json format output contain this non human readable header string:
+    no CamelCase and no space.
     """
+    return text.lower().replace(' ', '_')
+
+
+def guideline_rules_for_checker(checker, context):
+    """
+    Returns the guideline and rules covered by the given checker. This
+    function returns a dict which maps guideline names to the covered
+    rules: { "sei-cert": ["arr30-c", ...], ... }.
+    """
+    result = defaultdict(list)
+    labels = context.checker_labels.labels_of_checker(checker)
+    guidelines = context.checker_labels.get_description('guideline')
+
+    for label in labels:
+        if label[0] in guidelines:
+            result[label[0]].append(label[1])
+
+    return result
 
-    # If the given output format is not 'table', redirect logger's output to
-    # the stderr.
-    logger.setup_logger(args.verbose if 'verbose' in args else None,
-                        None if args.output_format == 'table' else 'stderr')
 
+def __get_detailed_checker_info(args, cl):
```

Comment:
Docstrings are missing for newly introduced functions and also add type hints where it is possible in this file. Please also do this everywhere in this file where you introduced a new function.